pytlu/online_monitor/pytlu_converter.py

```diff
@@ -0,0 +1,125 @@
+                        return None
+            except AttributeError:  # Happens if first data is not meta data
+                return None
+        return {'meta_data': self.meta_data}
+
+    def interpret_data(self, data):
+        # add function to fill arrays with data and shift through
+        def fill_arrays(array, data, time, time_index):
+            array[0][time_index] = time
+            array[1] = np.roll(array[1], 1)
+            array[1][0] = data
+            return array
+
+        if data[0][1] is not None:
+            meta_data = data[0][1]['meta_data']
+            data_length = meta_data['data_length']
+            timestamp_start = meta_data['timestamp_start']
+            timestamp_stop = meta_data['timestamp_stop']
+            self.status_data['trigger_rate'] = data_length / (timestamp_stop - timestamp_start) / 1e3  # trigger rate in kHz
+
+            # fill time and data axes, here only one key (trigger rate) up to now
+            for key in self.all_arrays:
+                # update starting time (self.shift_cycle_time) if we just started or once shifted through the data array
+                if self.array_indices[key] == 0 or self.array_indices[key] % self.array_size[1] == 0:
+
+                    self.shift_cycle_times[key] = meta_data['timestamp_start']
+                    self.update_time_indices[key] = 0
+
+                # time since we started or last shifted through
+                self.now[key] = self.shift_cycle_times[key] - meta_data['timestamp_start']
+
+                self.all_arrays[key] = fill_arrays(self.all_arrays[key], self.status_data[key][0], self.now[key], self.update_time_indices[key])
+
+                # increase indices for timing and data
+                self.update_time_indices[key] += 1
+                self.array_indices[key] += 1
+
+            now = float(meta_data['timestamp_stop'])
+            recent_fps = 1.0 / (now - self.updateTime)  # calculate FPS
+            self.updateTime = now
+            self.fps = self.fps * 0.7 + recent_fps * 0.3
+
+            return [{'tlu': self.all_arrays, 'indices': self.array_indices, 'fps': self.fps, 'timestamp_stop': now}]
+
+        self.readout += 1
+
+        if self.n_readouts != 0:  # = 0 for infinite integration
+            if self.readout % self.n_readouts == 0:
+                self.trigger_rate_array = np.zeros_like(self.trigger_rate_array)
+                self.array_indices['trigger_rate'] = 0
+                self.update_time_indices['trigger_rate'] = 0
+                self.readouts = 0
+
+    def serialze_data(self, data):
+        return jsonapi.dumps(data, cls=utils.NumpyEncoder)
+
+    def handle_command(self, command):
+        # received signal is 'ACTIVETAB tab' where tab is the name (str) of the selected tab in online monitor
+        if 'ACTIVETAB' in command[0]:
+            self.active_tab = str(command[0].split()[1])
```

Comment:
Oh yes, thats true. I removed it because it was not working out of the box.. I think the name of the tab was simply wrong. Okay, but since the analysis of the plotted data is so simply we dont need this feature?

Reply:
Yes. You want to count triggers even when you do not look at the GUI tab. The active tab feature is rarely used/useful. The one use case that we have are online correlations that are computationally too demanding to do them always for all devices.

Reply:
okay ;)